python/ray/data/datasource/parquet_datasource.py

```diff
@@ -36,11 +36,16 @@
 PARQUET_READER_ROW_BATCH_SIZE = 100000
 FILE_READING_RETRY = 8
 
-# The estimated bytes size multiplier for reading Parquet data source in Arrow,
+# The minimum size multiplier for reading Parquet data source in Arrow,
 # as Arrow in-memory representation uses much more memory compared to Parquet
 # uncompressed representation. See https://github.com/ray-project/ray/pull/26516
-# for more context.
-PARQUET_TO_ARROW_SIZE_MULTIPLIER = 5
+# for more context. Datasets will try to estimate the correct multiplier, using this
+# constant as a lower bound for safety.
+PARQUET_COMPRESSION_RATIO_ESTIMATE_LOWER_BOUND = 5
+
+# The number of row samples to take from the dataset to try to get an estimate of the
+# parquet compression ratio.
+PARQUET_COMPRESSION_RATIO_ESTIMATE_NUM_SAMPLES = 4
```

Comment:
It's a good starting point. I am also thinking to introduce threshold based on ratio

```
# The sampling ratio for the dataset (i.e. precentage of row groups to sample)
PARQUET_COMPRESSION_RATIO_ESTIMATE_SAMPLES_RATIO = 0.04

# The maximal number of row samples to take from the dataset
PARQUET_COMPRESSION_RATIO_ESTIMATE_NUM_SAMPLES = 50
```

And we can optimize later to launch the sampling tasks remotely, if sampling takes too much time.

Reply:
Yeah, I think something like using tasks to sample makes a lot of sense. Doesn't going beyond, say 10-50 samples lead to diminishing returns regardless of the size of the dataset though? The variance of the estimate should be reduced based on the number of samples in the same way, regardless of dataset size.

Reply:
@ericl - yeah I agree, we need cap this o.w. it becomes wasteful like reading data twice.